src/rabbit_peer_discovery.erl

```diff
@@ -183,6 +204,34 @@ unregister() ->
       ok
   end.
 
+-spec lock() -> ok | {ok, Data :: term()} | not_supported | {error, Reason :: string()}.
+
+lock() ->
+    Backend = backend(),
+    rabbit_log:info("Will try to lock with peer discovery backend ~s", [Backend]),
+    case Backend:lock(node()) of
```

Comment:
There was a missing PR: https://github.com/rabbitmq/rabbitmq-common/pull/204/files.

The `lock/1` return value of `unsupported` is not very consistent with `supports_registration/0` but
it's consistent with rabbitmq-autocluster :/. Arguably this is not a show stopper for #1258 and 3.7.0 release plans.